src/utils/lollipop/crypto.ts

```diff
@@ -0,0 +1,70 @@
+import * as crypto_lib from "crypto";
+import { verifySignatureHeader } from "@mattrglobal/http-signatures";
+import * as TE from "fp-ts/TaskEither";
+import * as H from "@pagopa/handler-kit";
+import { JwkPublicKey } from "@pagopa/ts-commons/lib/jwk";
+import { pipe } from "fp-ts/lib/function";
+import * as E from "fp-ts/Either";
+import * as jose from "jose";
+import { AssertionRef } from "../../generated/definitions/fn-lollipop/AssertionRef";
+import { JwkPubKeyHashAlgorithm } from "../../types/lollipop";
+import { getCustomVerifyWithEncoding } from "./httpSignature.verifiers";
+
+export const validateHttpSignatureWithEconding = (
+  dsaEncoding: crypto_lib.DSAEncoding
+) => (
+  request: H.HttpRequest,
+  assertionRef: AssertionRef,
+  publicKey: JwkPublicKey,
+  body?: string
+): TE.TaskEither<Error, true> =>
```

Comment:
We could move to object param

```suggestion
export const validateHttpSignatureWithEconding = (
  dsaEncoding: crypto_lib.DSAEncoding
) => ({
  request,
  assertionRef,
  publicKey,
  body
}: {
  readonly request: H.HttpRequest;
  readonly assertionRef: AssertionRef;
  readonly publicKey: JwkPublicKey;
  readonly body?: string;
}): TE.TaskEither<Error, true> =>
```

Reply:
Fixed in 28aacc1